### PR TITLE
many: add shellcheck to static tests

### PR DESCRIPTION
**Pull request by kyrofa**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

Also ensure all scripts pass. **Note:** part of the modification here involves autopkgtests, so we should probably run those once Travis is green.

### CODE REVIEW

File: bin/snapcraftctl [hunk 1]
```diff
@@ -14,17 +14,29 @@
 # snapcraft). To that end, this script will use the interpreter defined within
 # the SNAPCRAFT_INTERPRETER environment variable.
 
+# Shell quote arbitrary string by replacing every occurrence of '
```
Comment (sergiusens) [inline bin/snapcraftctl]: How is this related to shellcheck?

File: demos/gradle/gradlew [hunk 2]
```diff
@@ -19,12 +20,17 @@ while [ -h "$PRG" ] ; do
         PRG=`dirname "$PRG"`"/$link"
     fi
 done
+# shellcheck disable=SC2006
```
Comment (sergiusens) [inline demos/gradle/gradlew]: if you are going to add these comments, why not just change them to `$()`?

Comment (sergiusens) [reply demos/gradle/gradlew]: Unless there is a way to skip the whole file without modifying it.

Comment (kyrofa) [reply demos/gradle/gradlew]: This file was automatically generated, so I'd rather not actually try to fix all its numerous issues or I'll have to pretend I understand what this script is doing and test it :stuck_out_tongue: . I'll hard-code a skip for it.

Comment (kyrofa) [reply demos/gradle/gradlew]: Done.
